core-drawer-panel.html

```diff
@@ -75,344 +75,425 @@
 <link rel="import" href="../core-selector/core-selector.html">
 
-<polymer-element name="core-drawer-panel" touch-action="auto">
-<template>
-
+<dom-module id="core-drawer-panel">
   <link rel="stylesheet" href="core-drawer-panel.css">
 
-  <core-media-query query="max-width: {{forceNarrow ? '' : responsiveWidth}}" queryMatches="{{queryMatches}}"></core-media-query>
+  <template>
+    <core-media-query query="[[mediaQuery]]" on-query-matches-changed="onQueryMatchesChanged"></core-media-query>
+
+    <core-selector class-name="[[coreSelectorClass]]" attrForSelected="id" selected="{{selected}}">
 
-  <core-selector class="{{ {'narrow-layout' : narrow, transition : transition, dragging : dragging, 'right-drawer': rightDrawer} | tokenList }}" valueattr="id" selected="{{selected}}">
+      <div id="main" style="[[mainStyle]]">
+        <content select="[main]"></content>
+        <div id="scrim" on-tap="[[togglePanel]]"></div>
+        <div id="edgeSwipeOverlay" class-name="[[swipeOverlayClass]]"></div>
+      </div>
 
-    <div id="main" _style="left: {{ narrow || rightDrawer ? '0' : drawerWidth }}; right: {{ rightDrawer ? (narrow ? '' : drawerWidth) : '' }};">
-      <content select="[main]"></content>
-      <div id="scrim" on-tap="{{togglePanel}}"></div>
-      <div id="edgeSwipeOverlay" hidden?="{{!narrow || disableEdgeSwipe}}"></div>
-    </div>
+      <div id="drawer" style="[[drawerStyle]]">
+        <content select="[drawer]"></content>
+      </div>
 
-    <div id="drawer" _style="width: {{ drawerWidth }}">
-      <content select="[drawer]"></content>
-    </div>
+    </core-selector>
+  </template>
 
-  </core-selector>
+</dom-module>
 
-</template>
 <script>
 
-  Polymer('core-drawer-panel', {
-
-    /**
-     * Fired when the narrow layout changes.
-     *
-     * @event core-responsive-change
-     * @param {Object} detail
-     * @param {boolean} detail.narrow true if the panel is in narrow layout.
-     */
-     
-    /**
-     * Fired when the selected panel changes.
-     * 
-     * Listening for this event is an alternative to observing changes in the `selected` attribute.
-     * This event is fired both when a panel is selected and deselected.
-     * The `isSelected` detail property contains the selection state.
-     * 
-     * @event core-select
-     * @param {Object} detail
-     * @param {boolean} detail.isSelected true for selection and false for deselection
-     * @param {Object} detail.item the panel that the event refers to
-     */
-
-    publish: {
+  (function() {
 
-      /**
-       * Width of the drawer panel.
-       *
-       * @attribute drawerWidth
-       * @type string
-       * @default '256px'
-       */
-      drawerWidth: '256px',
+    'use strict';
 
-      /**
-       * Max-width when the panel changes to narrow layout.
-       *
-       * @attribute responsiveWidth
-       * @type string
-       * @default '640px'
-       */
-      responsiveWidth: '640px',
+    function classNames(obj) {
```

Comment:
I made the same comment in core-toolbar: can we put this somewhere centralized / shareable?